### PR TITLE
added evancook.audio tenpad macropad

### PR DESCRIPTION
**Pull request by evanmcook**
## Description

I have added a keyboard, evancook.audio tenpad. tenpad was designed with the goal of compactness in the z direction, while still using cherry MX keyswitches

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

(NB: i am 95% sure that my work follows the C guide, but I am not a confident coder at this juncture. i thank the reviewers for their patience!)

### CODE REVIEW

File: keyboards/evancook.audio/tenpad/config.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * 
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x23F8
+#define PRODUCT_ID      0x10p4
```
Comment (drashna) [inline keyboards/evancook.audio/tenpad/config.h:24]: This needs to be a hex value. "p" is not a valid hex digit.

File: keyboards/evancook.audio/tenpad/config.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * 
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x23F8
```
Comment (drashna) [inline keyboards/evancook.audio/tenpad/config.h:23]: This Vendor ID is in use: 
https://www.the-sz.com/products/usbid/index.php?v=0x23F8&p=&n=

Please change it.  Anything about 0x4000 should be good, but the higher, the better.

File: keyboards/evancook.audio/tenpad/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/evancook.audio/tenpad/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/evancook.audio/tenpad/rules.mk [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = no  # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/evancook.audio/tenpad/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
```

File: keyboards/evancook.audio/tenpad/rules.mk [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+MOUSEKEY_ENABLE  = no  # Mouse keys
+EXTRAKEY_ENABLE  = no  # Audio control and System control
+CONSOLE_ENABLE   = yes # Console for debug
+COMMAND_ENABLE   = yes # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
```
Comment (drashna) [inline keyboards/evancook.audio/tenpad/rules.mk:15]:
```suggestion
```

File: keyboards/evancook.audio/tenpad/rules.mk [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+NKRO_ENABLE      = no  # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/evancook.audio/tenpad/rules.mk:16]:
```suggestion
NKRO_ENABLE      = no  # Enable N-Key Rollover
```

File: keyboards/evancook.audio/tenpad/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+BACKLIGHT_ENABLE = no # Enable keyboard backlight functionality
+AUDIO_ENABLE     = no
+RGBLIGHT_ENABLE  = no
+OLED_DRIVER_ENABLE = no
```
Comment (drashna) [inline keyboards/evancook.audio/tenpad/rules.mk:20]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Review by drashna — APPROVED** [review]

File: keyboards/evancook.audio/tenpad/rules.mk [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE  = no  # Mouse keys
+EXTRAKEY_ENABLE  = no  # Audio control and System control
+CONSOLE_ENABLE   = yes # Console for debug
+COMMAND_ENABLE   = yes # Commands for debug and configuration
+NKRO_ENABLE      = no  # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no # Enable keyboard backlight functionality
+AUDIO_ENABLE     = no
+RGBLIGHT_ENABLE  = no
```
Comment (fauxpark) [inline keyboards/evancook.audio/tenpad/rules.mk:18]:
```suggestion
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/evancook.audio/tenpad/readme.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# tenpad macropad
+
+A 2x5 macropad designed by Evan Cook, with the design goal of being compact in the Z axis
+[Default Layout](https://raw.githubusercontent.com/evanmcook/tenpad/main/keyboard-layout.jpg)
+
+Keyboard Maintainer: [Evan Cook](https://github.com/evanmcook/)  
+Hardware Supported: tenpad Macropad PCB  
+Hardware Availability: [tenpad Github](https://github.com/evanmcook/tenpad)
```
Comment (fauxpark) [inline keyboards/evancook.audio/tenpad/readme.md:8]:
```suggestion
* Keyboard Maintainer: [Evan Cook](https://github.com/evanmcook/)
* Hardware Supported: tenpad Macropad PCB
* Hardware Availability: [tenpad Github](https://github.com/evanmcook/tenpad)
```

File: keyboards/evancook.audio/tenpad/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+ * |   Q   |   W   |   E   |   R   |   T   |
+ * |-------+-------+-------|-------|-------|
+ * |   A   |   S   |   D   |   F   |   G   |
+ * ,---------------------------------------.
+ * 
+ * 
+ */
+[0] = LAYOUT( \
+  KC_Q,       KC_W,      KC_E,      KC_R,      KC_T,         \
+  KC_A,       KC_S,      KC_D,      KC_F,      KC_G         \
```
Comment (fauxpark) [inline keyboards/evancook.audio/tenpad/keymaps/default/keymap.c:31]:
```suggestion
[0] = LAYOUT(
  KC_Q,       KC_W,      KC_E,      KC_R,      KC_T,
  KC_A,       KC_S,      KC_D,      KC_F,      KC_G
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
`.` is not allowed in keyboard folder names.

**Comment by evanmcook** [comment]
thank you all for your patience with me, i greatly appreciate it.

File: keyboards/evancookaudio/tenpad/readme.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+A 2x5 macropad designed by Evan Cook, with the design goal of being compact in the Z axis
+[Default Layout](https://raw.githubusercontent.com/evanmcook/tenpad/main/keyboard-layout.jpg)
+
+* Keyboard Maintainer: [Evan Cook](https://github.com/evanmcook/)
+* Hardware Supported: tenpad Macropad PCB
+* Hardware Availability: [tenpad Github](https://github.com/evanmcook/tenpad)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make tenpad:default
```
Comment (fauxpark) [inline keyboards/evancookaudio/tenpad/readme.md:12]:
```suggestion
    make evancookaudio/tenpad:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Missing an info.json.

**Comment by evanmcook** [comment]
apologies for requesting a review preemptively to knowing my code will pass muster. i cannot find a resource that clarifies the error that returns from running qmk lint. 

`Invalid API data: evancookaudio/tenpad: usb.device_ver: '0x0' does not match '^0x[0-9A-F]{4}$'`

is this related to usb device version information in info.json?

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**